### PR TITLE
Baidu search suggestion

### PR DESCRIPTION
**Pull request by jiaojianhua**
Add Baidu search suggestion.
commit: 	ad9ef5d
please check the save.txt of diff
[save.txt](https://github.com/anthonycr/Lightning-Browser/files/1041630/save.txt)

### CODE REVIEW

**Comment by jiaojianhua** [comment]
Clear code in: eda7fe0

File: app/src/main/res/xml/preference_general.xml [hunk 1]
```diff
@@ -26,18 +26,9 @@
...
-        <Preference
             android:key="download"
             android:title="@string/download" />
         <Preference
-            android:key="home"
-            android:title="@string/home" />
-        <Preference
-            android:key="search"
-            android:title="@string/search" />
-        <Preference
```
Comment (anthonycr) [inline app/src/main/res/xml/preference_general.xml]: Please add back all these preferences

File: app/src/main/java/acr/browser/lightning/fragment/GeneralSettingsFragment.java [hunk 2]
```diff
@@ -161,22 +164,22 @@ private void initPrefs() {
...
+//            R.string.action_ok,
+//            new BrowserDialog.EditorListener() {
+//                @Override
+//                public void onClick(String text) {
+//                    mPreferenceManager.setSearchUrl(text);
+//                    searchengine.setSummary(mActivity.getString(R.string.custom_url) + ": " + text);
+//                }
+//            });
+//
+//    }
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/fragment/GeneralSettingsFragment.java]: Removing the option to change the search engine is not acceptable

File: app/src/main/java/acr/browser/lightning/fragment/GeneralSettingsFragment.java [hunk 3]
```diff
@@ -285,82 +288,82 @@ public void onClick(DialogInterface dialogInterface, int i) {
...
+//                        case 3:
+//                            homePicker();
+//                            break;
+//                    }
+//                }
+//            });
+//        picker.setPositiveButton(getResources().getString(R.string.action_ok), null);
+//        Dialog dialog = picker.show();
+//        BrowserDialog.setDialogSize(mActivity, dialog);
+//    }
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/fragment/GeneralSettingsFragment.java]: Removing the option to change the homepage and search option is not acceptable

File: app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java [hunk 4]
```diff
@@ -224,12 +225,12 @@ public boolean getSavePasswordsEnabled() {
...
 
     public int getSearchChoice() {
-        return mPrefs.getInt(Name.SEARCH, 1);
+        return mPrefs.getInt(Name.SEARCH, 0);
     }
 
     @NonNull
     public String getSearchUrl() {
-        return mPrefs.getString(Name.SEARCH_URL, Constants.GOOGLE_SEARCH);
+        return mPrefs.getString(Name.SEARCH_URL, Constants.BAIDU_SEARCH);
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java]: Please revert to back to google

File: app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java [hunk 5]
```diff
@@ -166,7 +167,7 @@ public boolean getHideStatusBarEnabled() {
 
     @NonNull
     public String getHomepage() {
-        return mPrefs.getString(Name.HOMEPAGE, Constants.SCHEME_HOMEPAGE);
+        return mPrefs.getString(Name.HOMEPAGE, Constants.HOME_PAGE_URL);
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java]: Please revert back to `SCHEME_HOMEPAGE`

File: app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java [hunk 6]
```diff
@@ -224,12 +225,12 @@ public boolean getSavePasswordsEnabled() {
     }
 
     public int getSearchChoice() {
-        return mPrefs.getInt(Name.SEARCH, 1);
+        return mPrefs.getInt(Name.SEARCH, 0);
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java]: Please revert

File: app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java [hunk 7]
```diff
@@ -104,7 +105,7 @@ public void setBookmarkAndTabsSwapped(boolean swap) {
     }
 
     public boolean getAdBlockEnabled() {
-        return mPrefs.getBoolean(Name.BLOCK_ADS, false);
+        return mPrefs.getBoolean(Name.BLOCK_ADS, true);
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java]: Please revert back to original value

File: app/src/main/java/acr/browser/lightning/constant/Constants.java [hunk 8]
```diff
@@ -31,6 +31,10 @@ private Constants() {
     public static final String BAIDU_SEARCH = "https://www.baidu.com/s?wd=";
     public static final String YANDEX_SEARCH = "https://yandex.ru/yandsearch?lr=21411&text=";
 
+    // Home page url
+    public static final String HOME_PAGE_URL = "http://zhadan.me/";
+    public static final String HOME_DOMAIN = "zhadan.me";
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/constant/Constants.java]: Please remove these

File: ChangeLog.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+##0.0.1(未发布)
```
Comment (anthonycr) [inline ChangeLog.md]: Please remove this file

File: app/src/main/java/acr/browser/lightning/fragment/GeneralSettingsFragment.java [hunk 10]
```diff
@@ -48,10 +48,10 @@
     private Activity mActivity;
     private static final int API = android.os.Build.VERSION.SDK_INT;
     private CharSequence[] mProxyChoices;
-    private Preference proxy, useragent, downloadloc, home, searchengine, searchsSuggestions;
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/fragment/GeneralSettingsFragment.java]: Please revert all the option removals in this class

File: app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java [hunk 11]
```diff
@@ -267,7 +268,7 @@ public int getProxyChoice() {
     }
 
     public int getUserAgentChoice() {
-        return mPrefs.getInt(Name.USER_AGENT, 1);
+        return mPrefs.getInt(Name.USER_AGENT, 3);
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/preference/PreferenceManager.java]: Please change back to the original value

File: app/src/main/java/acr/browser/lightning/search/BaiduSuggestionsModel.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+import acr.browser.lightning.R;
+import acr.browser.lightning.database.HistoryItem;
+import acr.browser.lightning.utils.FileUtils;
+
+// http://unionsug.baidu.com/su?wd=encodeURIComponent(U)
+// http://suggestion.baidu.com/s?wd=encodeURIComponent(U)&action=opensearch
+
+
+class BaiduSuggestionsModel extends BaseSuggestionsModel {
```
Comment (anthonycr) [inline app/src/main/java/acr/browser/lightning/search/BaiduSuggestionsModel.java]: Nice!!

**Comment by jiaojianhua** [comment]
merge to another branch for PR #577.
Please use #577 to do PR. 

Thanks

**Pull request closed without merging**